suite-native/device-manager/src/components/DeviceManagerModal.tsx

```diff
@@ -0,0 +1,45 @@
+import { Modal } from 'react-native';
+import { ReactNode } from 'react';
+import { useSafeAreaInsets, EdgeInsets } from 'react-native-safe-area-context';
+
+import { Box, ScreenHeaderWrapper } from '@suite-native/atoms';
+import { prepareNativeStyle, useNativeStyles } from '@trezor/styles';
+import { BlurredScreenOverlay } from '@suite-native/screen-overlay';
+
+import { DeviceSwitch } from './DeviceSwitch';
+import { useDeviceManager } from '../hooks/useDeviceManager';
+
+type DeviceManagerModalProps = {
+    children: ReactNode;
+};
+
+const contentWrapper = prepareNativeStyle<{ insets: EdgeInsets }>((utils, { insets }) => ({
+    paddingTop: Math.max(insets.top, utils.spacings.medium),
+    backgroundColor: utils.colors.backgroundSurfaceElevation0,
+}));
+
+export const DeviceManagerModal = ({ children }: DeviceManagerModalProps) => {
+    const { applyStyle } = useNativeStyles();
+
+    const insets = useSafeAreaInsets();
+
+    const { setIsDeviceManagerVisible, isDeviceManagerVisible } = useDeviceManager();
+
+    const handleClose = () => {
+        setIsDeviceManagerVisible(false);
+    };
+
+    return (
+        <Modal onRequestClose={handleClose} visible={isDeviceManagerVisible}>
+            <BlurredScreenOverlay />
+            <Box style={applyStyle(contentWrapper, { insets })}>
+                <ScreenHeaderWrapper>
+                    <Box flexDirection="row">
```

Comment:
nit: unnecessary Box? Or if is the `flexDirection` needed, It should be IMO inside of the `DeviceSwitch`